packages/react-components/react-button/src/components/Button/renderButton.tsx

```diff
@@ -3,19 +3,23 @@
 
 import { assertSlots } from '@fluentui/react-utilities';
 import type { ButtonSlots, ButtonState } from './Button.types';
+import { InProgress, Success } from '../common';
 
 /**
  * Renders a Button component by passing the state defined props to the appropriate slots.
  */
 export const renderButton_unstable = (state: ButtonState) => {
   assertSlots<ButtonSlots>(state);
-  const { iconOnly, iconPosition } = state;
+  const { iconOnly, iconPosition, actionState } = state;
 
   return (
     <state.root>
-      {iconPosition !== 'after' && state.icon && <state.icon />}
+      {actionState === 'inprogress' && (iconOnly || !state.icon) && <InProgress state={state} />}
```

Comment:
Even though we are using official Spinner components to compose this component, it would be nice to provide a way to override the InProgress and Success icons.

Reply:
Yes, user can use custom icons, by using icon slot. (Added this info in the description.)